src/mixins/legal-api-mixin.ts

```diff
@@ -41,26 +48,33 @@ export default class LegalApiMixin extends Vue {
 
   /**
    * Fetches a draft filing.
-   * @returns a promise to return the filing
+   * @returns a promise to return the draft filing, or null if not found
    */
   async fetchDraft (): Promise<any> {
     // get the draft filing from the tasks endpoint
     const url = `businesses/${this.getBusinessIdentifier}/tasks`
     return axios.get(url)
-      .then(res => {
+      .then(response => {
         // look at only the first task
-        const filing = res?.data?.tasks[0]?.task?.filing
-        const filingId = +filing?.header?.filingId
-        if (!filing || !filingId) {
-          throw new Error('invalid API response')
+        const todoName = response?.data?.tasks[0]?.task?.todo?.header?.name
+        const filing = response?.data?.tasks[0]?.task?.filing
+        const filingName = filing?.header?.name
+        const filingId = +filing?.header?.filingId // may be NaN
+        if (todoName === this.NAME_REQUEST) {
+          return null // no draft (task is still a NR)
+        }
+        if (filingName !== this.INCORPORATION_APPLICATION || !filingId) {
+          throw new Error('Invalid API response')
         }
         // save Filing ID from the header
         this.setFilingId(filingId)
         return filing
       })
-      .catch(() => {
-        // TODO: Throw a flag to the ui from here, if we want to trigger error handling in ui
-        return undefined // no draft found
+      .catch((error) => {
+        if (error?.response?.status === NOT_FOUND) {
+          return null // NR not found
+        }
+        throw error
```

Comment:
This code will have to be revised when we use the Temporary Registration Number.
- Will the API still respond with a todo task vs filing task?
- Will the API respond with NOT_FOUND?

The code above works with the current API (as of today, now).

Reply:
Imagine the length of the short circuiting code if we didn't have optional chaining!

Looks good, we will have to cross that bridge when we come to it :)